### PR TITLE
PR template - add a reviewer guide

### PR DESCRIPTION
**Pull request by laurenfrederick**
## Description

Add a Reviewer Checklist to the PR template (this template!) to guide reviewers and make sure when reviewing, all the pertinent pieces of a review are hit.

## Issue number or link

NA

## Author checklist

- [ ] I have submitted a [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) (only needed once). - Do I need to do this?
- [X] I have done a full self-review of my code. - No code
- [ ] I have updated `CHANGES.md` with a short summary of my change (for user-facing changes). - These changes do not seem to be user facing 
- ~~[ ] I have added or updated unit tests to ensure consistent code coverage as necessary.~~ No code
- ~~[ ] I have updated the documentation as necessary.~~ - N/A

## Remaining Tasks

NA 

## Testing plan

NA

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -39,4 +39,18 @@ To ensure your Pull Request is reviewed and accepted quickly, please refer to ou
...
+<!-- Include links to any required data or screenshots. Mention any edge cases such as user error, invalid data, etc. -->
+
+## Reviewer checklist
+
+Thank you for taking the time to review this PR. By approving a PR you are taking as much responsibility for these changes as the author.
+
+As you review, please go through the checklist below:
+
+- [ ] Review and run all parts of the test plan on this branch and verify it matches expectations.
+- [ ] Review the code and make sure you do not have any remaining questions or concerns. You should understand the code change and with the approach. If you are not confident or have doubts about the code please do not hesitate to ask questions.
```
Comment (j9liu) [inline .github/pull_request_template.md]: Typo?

```suggestion
- [ ] Review the code and make sure you do not have any remaining questions or concerns. You should understand the code change and the chosen approach. If you are not confident or have doubts about the code, please do not hesitate to ask questions.
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -39,4 +39,18 @@ To ensure your Pull Request is reviewed and accepted quickly, please refer to ou
...
+    - [ ] Code should follow the [C++ Style Guide](https://cesium.com/learn/cesium-native/ref-doc/style-guide.html)
+- [ ] Review the unit tests and make sure there are no missing tests or edge cases.
+- [ ] If the issue is a bug please make sure you can reproduce the bug in the main branch and then checkout this branch to make sure it actually solved the issue.
```
Comment (j9liu) [inline .github/pull_request_template.md]: **NITPICK:** Move as a sub-bullet under the first checkbox, since it's related?

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -39,4 +39,18 @@ To ensure your Pull Request is reviewed and accepted quickly, please refer to ou
...
+- [ ] Review documentation changes and updates to `CHANGES.md` to make sure they accurately cover the work in this PR
+- [ ] Review that the [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) has been submitted, if needed
```
Comment (j9liu) [inline .github/pull_request_template.md]: **NITPICK**:

```suggestion
- [ ] Review documentation changes and updates to `CHANGES.md` to make sure they accurately cover the work in this PR.
- [ ] Verify that the [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) has been submitted, if needed.
```

File: .github/pull_request_template.md (hunk 1 above)
Comment (laurenfrederick) [reply .github/pull_request_template.md]: Yep, thank you!

File: .github/pull_request_template.md (hunk 2 above)
Comment (laurenfrederick) [reply .github/pull_request_template.md]: Good idea - done!

**Comment by laurenfrederick** [comment]
@j9liu Ready for re-review. Thank you!
